intel_extension_for_transformers/llm/runtime/graph/models/model_utils/model_utils.cpp

```diff
@@ -80,16 +82,42 @@ static bool kv_cache_init(const struct model_hparams& hparams, struct model_kv_c
 
   // NE_TYPE_JBLAS can not be allocated memory
   const auto wtype_alloc = wtype == NE_TYPE_JBLAS ? NE_TYPE_I8 : wtype;
-  cache.k = ne_new_tensor_1d(cache.ctx, wtype_alloc, n_elements_k + NE_ALIGNMENT, NE_SIZE_CALC);
-  cache.k = ne_view_1d(cache.ctx, cache.k, n_elements_k,
-                       NE_ALIGNMENT - (reinterpret_cast<uintptr_t>(cache.k->data) % NE_ALIGNMENT));
-  cache.k->type = wtype;
-  cache.v = ne_new_tensor_1d(cache.ctx, wtype_alloc, n_elements_v + NE_ALIGNMENT, NE_SIZE_CALC);
-  cache.v = ne_view_1d(cache.ctx, cache.v, n_elements_v,
-                       NE_ALIGNMENT - (reinterpret_cast<uintptr_t>(cache.v->data) % NE_ALIGNMENT));
-  cache.v->type = wtype;
-  ne_set_name(cache.k, "cache_k");
-  ne_set_name(cache.v, "cache_v");
+
+  if (model) {  // non-null param of model for chatglm style kv-cache
```

Comment:
what is ```chatglm style```

Reply:
Allocating KV-cache as components of each layer rather than as 2 huge chunks for all layers. I'll explain more in the comments.

Reply:
Updated in 2f4c3ca